io/disk/arcconf/arcconf_drive_oper.py

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/env python
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+#
+# Copyright: 2016 IBM
+# Author: Manvanthara B Puttashankar <manvanth@linux.vnet.ibm.com>
+
+
+"""
+arcconf - Array configuration utility for PMC-Sierra
+(Microsemi) Controllers.
+"""
+
+from avocado import Test
+from avocado.utils.software_manager import SoftwareManager
+from avocado.utils import process
+from avocado import main
+from avocado.utils import distro
+
+
+class Arcconftest(Test):
+    """
+    Covers functionality relavent to device settings.
+    """
+
+    def setUp(self):
+
+        """
+        Checking if the required packages are installed,
+        if not found specific packages will be installed.
+        """
+        smm = SoftwareManager()
+        if not smm.check_installed("lsscsi") and not smm.install("lsscsi"):
+            self.skip("Unable to install lsscsi")
+        self.crtl_no = self.params.get('crtl_no')
+        self.channel_no = self.params.get('channel_no')
+        self.disk_no = self.params.get('disk_no', default="").split(",")
+        self.pci_id = self.params.get('pci_id', default="").split(",")
+        self.logicaldrive = self.params.get('logicaldrive')
+        self.diskoverwrite = self.params.get('diskoverwrite')
+        self.http_path = self.params.get('http_path')
+        self.tool_name = self.params.get('tool_name')
+        self.option = self.params.get("option")
+        self.option_args = self.params.get("option_args")
+        self.option_args2 = self.params.get("option_args2")
+
+        # Gets the list of PCIIDs on the system
+        cmd = 'for device in $(lspci | awk \'{print $1}\') ; do echo \
+              $(lspci -vmm -nn -s $device | grep "\[" | awk \'{print $NF}\' \
+              | sed -e "s/\]//g" | sed -e "s/\[//g" | tr \'\\n\' \' \' \
+              | awk \'{print $2,$3,$4,$5}\') ; done'
+        pci_id_formatted = []
+        for i in self.cmdop_list(cmd).splitlines():
+            pci_id_formatted.append(str(i.replace(" ", ":")))
+
+        # check if all the yaml parameters are entered
+        if self.crtl_no is '' or self.channel_no is '' or self.disk_no is \
+           '' or self.pci_id is '' or len(self.disk_no) <= 1 or \
+           self.tool_name is '' or self.http_path is '':
```

Comment:
@narasimhan-v , its on purpose.    the first one looks for any empty string and the second one looks for the miminal disk count.